### PR TITLE
feat: add CentOS Stream 10

### PR DESCRIPTION
**Pull request by burnsjared0415**
# 

## Summary of Pull Request

Adds CentOS Stream 10 to project.

## Type of Pull Request

- [ ] This is a bugfix. `type/bug`
- [x] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.
      Please describe:

## Related to Existing Issues

Issue Number: N/A

## Test and Documentation Coverage

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

## Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: docs/getting-started/downloads.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Download the guest operating system ISOs using the download script (`./download.
 | :simple-oracle: Oracle Linux @span                          | 9         | :octicons-check-circle-16:{ .green }                                  |
 |                                                             | 8         | :octicons-check-circle-16:{ .green }                                  |
 | :fontawesome-brands-centos: CentOS @span                    | 9 Stream  | :octicons-check-circle-16:{ .green }                                  |
-|                                                             | 7         | :octicons-check-circle-16:{ .green }                                  |
+|                                                             | 10 Stream | :octicons-check-circle-16:{ .green }                                  |
```
Comment (tenthirtyam) [inline docs/getting-started/downloads.md]: Can you put 10 before 9?

File: docs/index.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ This project supports the following guest operating systems:
 | :simple-oracle: Oracle Linux @span                         | 9         | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
 |                                                            | 8         | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
 | :fontawesome-brands-centos: CentOS @span                   | 9 Stream  | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
+|                                                            | 10 Stream | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
```
Comment (tenthirtyam) [inline docs/index.md]: Can you put 10 before 9?

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
The README.md will need an update as will the GitLab CI files.

File: docs/getting-started/downloads.md (hunk 1 above)
Comment (burnsjared0415) [reply docs/getting-started/downloads.md]: done

File: docs/index.md (hunk 2 above)
Comment (burnsjared0415) [reply docs/index.md]: done

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,8 @@
 
 - Adds confirmation before overwriting existing configuration files.
   [#987](https://github.com/vmware-samples/packer-examples-for-vsphere/pull/987)
+- Added Centos 10 stream to project.
```
Comment (tenthirtyam) [inline CHANGELOG.md]:
```suggestion
- Adds Centos Stream 10 to project.
```

File: docs/index.md [hunk 4]
```diff
@@ -32,7 +32,8 @@ This project supports the following guest operating systems:
...
 | :simple-oracle: Oracle Linux @span                         | 9         | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
 |                                                            | 8         | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
-| :fontawesome-brands-centos: CentOS @span                   | 9 Stream  | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
+| :fontawesome-brands-centos: CentOS @span                   | 10 Stream  | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
+|                                                            | 9 Stream | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
```
Comment (tenthirtyam) [inline docs/index.md]:
```suggestion
| :fontawesome-brands-centos: CentOS @span                   | 10 Stream | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
|                                                            | 9 Stream  | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} | :octicons-check-circle-16:{.green} |
```

Comment (tenthirtyam) [reply docs/index.md]: Alignment.

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Update here: https://github.com/burnsjared0415/packer-examples-for-vsphere/blob/17a9469485a6dd9339734c401acdfec9e9b9df8a/README.md?plain=1#L42

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
And upate here, plus remove CentOS 7: https://github.com/vmware-samples/packer-examples-for-vsphere/blob/4880b2c8d55499e00592499172f09aaf0d555c6e/build-ci.yaml#L109-L120

Then run gomplate to generate a new gitlab yml.

**Comment by tenthirtyam** [comment]
Resolved the needed changes in fa01fef.

**Review by tenthirtyam — APPROVED** [review]
LGTM!

**Pull request merged**